docs/maintainer/understanding_conda_forge/compilation-concepts.md

```diff
@@ -0,0 +1,148 @@
+  library.
+
+Include files are installed into the `include` directory or its subdirectories. They usually use
+`.h` or `.inc` suffix for C. For C++, sometimes names without a suffix are used to follow the
+standard library `#include` scheme.
+
+Static libraries are installed into the `lib` directory and carry an `.a` suffix on Unix, or `.lib`
+on Windows. On Unix, shared libraries are used directly for linking, and they are described in the
+[binaries](#binaries) section. On Windows, import libraries are used instead. They use `.lib` suffix
+like static libraries.
+
+Finally, packages often provide additional files that are used at build time to determine how to use
+the library in question. For example, these can include pkg-config files, CMake files, autotools
+macros.
+
+## Binaries
+
+The primary kind of artifacts produced by compiled programming languages are binaries. In this
+context, binaries mean files containing machine code. There are two main kind of binaries:
+
+1. Executables: programs that can be run directly by the user.
+2. Shared libraries: collections of compiled code that programs usually link to and therefore they
+   are loaded by programs at start time.
+3. Loadable modules: collections of compiled code that are loaded by programs at runtime (e.g.
+   plugins).
+
+On Unix platforms, executables don't feature any suffix, and common shells only start executables
+when their filename matches the specified command exactly. On Windows, executables commonly use
+`.exe` suffix, and shells account for that. For example, the Python executable will be named
+`python` on Unixes, and `python.exe` on Windows; in both cases, typing `python` will execute it.
+Executables are usually installed into the `bin` directory.
+
+Shared libraries use `lib` prefix on Unixes, and `.so` suffix, except for macOS where they use
+`.dylib` suffix instead. They are installed into the `lib` directory. They often include a version
+string to indicate compatibility between different library versions. If that is the case, the
+installed library usually consists of three files:
+
+- the actual library containing full version string, such as `lib{name}.so.1.2.3` or
+  `lib{name}.1.2.3.dylib`,
+- a partially versioned symbolic link, such as `lib{name}.so.1` or `lib{name}.1.dylib`,
+- an unversioned symbolic link, such as `lib{name}.so` or `lib{name}.dylib`.
+
+When building a new program, the linker uses the unversioned symbolic link to select the library
+to link to. From it, the linker reads the library `SONAME` (or install name, on macOS) that
+corresponds to the partially versioned name, and stores that in the resulting executable. When run,
+the executable loads the library using its partially versioned name. When the library is upgraded
+to a newer version that is backwards compatible (i.e. has the same partially versioned name), all
+executables start using the new version.
```

Comment:
On macOS, there's more checking. specificlly the compatibility version and current version stored in the binaries.